layers/vk_layer_utils.h

```diff
@@ -220,7 +220,7 @@ static inline bool IsImageLayoutReadOnly(VkImageLayout layout) {
 }
 
 static inline bool IsImageLayoutDepthReadOnly(VkImageLayout layout) {
-    constexpr std::array<VkImageLayout, 7> read_only_layouts = {
+    constexpr std::array<VkImageLayout, 4> read_only_layouts = {
```

Comment:
With C++17 you can now write this cleaner.

```suggestion
    constexpr std::array read_only_layouts = {
```